### PR TITLE
Removed rnix dependencies, as it has been deprecated

### PR DESCRIPTION
**Pull request by GoodGuyPat**
Fixes [rnix deprecation](https://github.com/nix-community/nixvim/issues/1244).

### CODE REVIEW

**Review by NickHu — APPROVED** [review]

**Comment by NickHu** [comment]
CI errors look like spurious errors to me, possibly to do with jobs getting cancelled because of new jobs being queued up in rapid succession due to you adding more commits.

Regardless, they have passed now.

**Review by NickHu — APPROVED** [review]

**Review by NickHu — APPROVED** [review]

**Review by traxys — CHANGES_REQUESTED** [review]

File: plugins/lsp/language-servers/default.nix [hunk 1]
```diff
@@ -539,11 +539,6 @@ with lib; let
       name = "pyright";
       description = "pyright for Python";
     }
-    {
```
[the remark this answers is not in the document]
Comment (GoodGuyPat) [reply plugins/lsp/language-servers/default.nix:542]: Tested it and doesn't break the local build. lgtm

File: plugins/lsp/default.nix [hunk 2]
```diff
@@ -79,7 +79,6 @@ in {
         "pylsp"
         "pylyzer"
         "pyright"
-        "rnix-lsp"
```
Comment (traxys) [inline plugins/lsp/default.nix:82]: Don't include this line, it is needed for the deprecation

**Review by traxys — CHANGES_REQUESTED** [review]

File: plugins/lsp/default.nix (hunk 2 above)
Comment (GoodGuyPat) [reply plugins/lsp/default.nix:82]: I am quite noob wrt github, so Iincluded the changes in a new commit. That line is not present in that commit.

Comment (GoodGuyPat) [reply plugins/lsp/default.nix:82]: Ah, gotcha.

**Review by traxys — APPROVED** [review]

**Pull request merged**